src/net_processing.cpp

```diff
@@ -1252,6 +1253,47 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
            (GetBlockProofEquivalentTime(*pindexBestHeader, *pindex, *pindexBestHeader, m_chainparams.GetConsensus()) < STALE_RELAY_AGE_LIMIT);
 }
 
+bool PeerManagerImpl::FetchBlock(NodeId id, const CBlockIndex& index)
+{
+    PeerRef peer = GetPeerRef(id);
+    if (peer == nullptr) return false;
```

Comment:
`peer` isn't used. You can remove it.

Reply:
This is the only place where we check that this peer actually exists (i.e. wasn't disconnected). I'll add a comment.

Reply:
Actually, you check below that a `CNodeState` exists, which can only happen if the peer isn't disconnected, which makes this check redundant.

Reply:
Ok, I'll remove it then.